docs/extension-maintainers.md

```diff
@@ -0,0 +1,128 @@
+
+```json
+{
+    "name": "xdebug/xdebug",
+    "php-ext": {
+        "extension-name": "xdebug"
+    }
+}
+```
+
+#### `priority`
+
+`priority` is currently not used, but will form part of the `ini` filename to
+control ordering of extensions, if the target platform uses it.
+
+#### `support-zts`
+
+Indicates whether the extension supports Zend Thread-Safe (ZTS) mode or not.
+
+#### `configure-options`
+
+This is a list of parameters that may be passed to the `./configure` command.
+Each item of the list is a JSON object with:
+
+ * `name`, the parameter name itself
+ * `description`, a helpful description of what the parameter does
+ * optionally, `needs-value`, a boolean to tell PIE whether the parameter is a
+   simple flag (typically used for `--enable-this-flag` type parameters), or if
+   the parameter should have a value specified (typically used for
+   `--with-library-path=...` type parameters, where a value must be given by
+   the end user)
+
+When an end user is installing an extension with PIE, they may specify any
+defined `configure-options` that are passed to `./configure`. For example, if
+an extension defines the following `composer.json`:
+
+```json
+{
+    "name": "myvendor/myext",
+    "php-ext": {
+        "extension-name": "myext",
+        "configure-options": [
+            {
+                "name": "enable-my-flag",
+                "description": "Should my flag be enabled",
+                "needs-value": false
+            },
+            {
+                "name": "with-some-lib",
+                "description": "Specify the path to some-lib",
+                "needs-value": true
+            }
+        ]
+    }
+}
+```
+
+Then the `pie build` or `pie install` commands may be invoked in the following
+ways to achieve the desired configuration:
+
+ * `pie install myvendor/myext`
+   * This will simply invoke `./configure` without any parameters
+ * `pie install myvendor/myext --enable-my-flag`
+   * This will invoke `./configure --enable-my-flag`
+ * `pie install myvendor/myext --with-some-lib=/path/to/somelib`
+   * This will invoke `./configure --with-some-lib=/path/to/somelib`
+ * `pie install myvendor/myext --enable-my-flag --with-some-lib=/path/to/somelib`
+   * This will invoke `./configure --enable-my-flag --with-some-lib=/path/to/somelib`
+
+Note that it is not possible for end users of PIE to specify configuration
+options that have not been defined in your extension's `configure-options`
+definition. Using the same example above `composer.json`, invoking PIE with
+an invalid option, such as `pie install myvendor/myext --something-else` will
+result in an error `The "--something-else" option does not exist.`.
+
+If an end user does not specify a flag defined in the `configure-options`
+definition, it will simply not be passed to `./configure`. There is no way to
+specify a default value in the `configure-options` definition.
+
+## Submit the extension to Packagist
```

Comment:
Could you add a note that package authors also need to tag a release for the extension to be installable until support for building from commit references is added? Otherwise, this may confuse users that are used to not needing a tag to test a package.

Reply:
AFAIK, you can already do this, or at least from `dev-master`: 
```
pie install xdebug/xdebug:dev-master
```

Reply:
Since #13 was fixed, it should be possible as mentioned. I'll check Mongo shortly, and if it's not working try to figure out why separately